### PR TITLE
import: support the '--file' option

### PR DESCRIPTION
**Pull request by vyloy**
This PR add the '--file' option for `dvc import`

### CODE REVIEW

File: dvc/command/imp.py [hunk 1]
```diff
@@ -62,4 +64,14 @@ def add_parser(subparsers, parent_parser):
...
     )
+    import_parser.add_argument(
+        "-f",
+        "--file",
+        help="Specify name of the stage file. It should be "
+        "either 'Dvcfile' or have a '.dvc' suffix (e.g. "
+        "'prepare.dvc', 'clean.dvc', etc) in order for "
+        "dvc to be able to find it later. By default "
+        "the first output basename + .dvc is used as "
+        "a stage filename.",
```
Comment (shcheklein) [inline dvc/command/imp.py]: 1. Do we enforce the file name somewhere? What would happen if I specify it w/o `.dvc`? We need to check that it does not allow creating stage files like this. Also, let's add a test for this.
2. There is not first or second output in the import case.

In general, I don't like this description. It's too verbose. I would just keep something like - `specify the name of the DVC file it generates`. All other details should go to the docs.

The same questions stay for the `dvc add`. As we agreed in private we do both (even though it's turned out `-f` is already supported by `dvc add` but not documented).

File: tests/test_import.py [hunk 2]
```diff
@@ -156,3 +156,23 @@ def test(self):
...
+class TestImportFilename(TestDvc):
+    def setUp(self):
+        super(TestImportFilename, self).setUp()
+        tmp_dir = self.mkdtemp()
+        self.external_source = os.path.join(tmp_dir, "file")
+        with open(self.external_source, "w") as fobj:
+            fobj.write("content")
+
+    def test(self):
+        ret = main(["import", "-f", "bar.dvc", self.external_source])
```
Comment (shcheklein) [inline tests/test_import.py]: We need to test that it support subdirectories properly.

Comment (vyloy) [reply tests/test_import.py]: Do we need to create the subdirectories when they are not exist?

Comment (shcheklein) [reply tests/test_import.py]: good question! I think it should behave the same way as `dvc add` and `dvc run`

**Comment by vyloy** [comment]
Sure.

File: tests/func/test_import.py [hunk 3]
```diff
@@ -156,3 +156,30 @@ def test(self):
...
+        os.remove("bar.dvc")
+
+        ret = main(["import", "--file", "bar.dvc", self.external_source])
+        self.assertEqual(0, ret)
+        self.assertTrue(os.path.exists("bar.dvc"))
+
+        os.remove("bar.dvc")
+        os.mkdir("sub")
+
+        ret = main(["import", "--file", "sub/bar.dvc", self.external_source])
```
Comment (efiop) [inline tests/func/test_import.py]: Oh, one more thing. Please don't hardcode paths like `sub/bar.dvc` and use `os.path.join()` instead. We are running on both windows and *nix, so your test on windows was not creating `bar.dvc` in `sub` but rather a sole file named `sub/bar.dvc`(on windows / is a valid char in the filename).

**Comment by efiop** [comment]
@vyloy Thank you! :slightly_smiling_face:

**Pull request merged**